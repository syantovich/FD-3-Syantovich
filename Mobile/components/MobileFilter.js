import React from 'react';
import {voteEvents} from "./events";

import "./MobileClient.css"

class MobileClient extends React.PureComponent {
    render() {
        console.log("MobileFilter render");

        return (
            <div>
                <input type="button" value="Все" id={"all"} onClick={()=>{voteEvents.emit("EfilterClientsMode","all")}}/>
                <input type="button" value="Активные" id={"active"} onClick={()=>{voteEvents.emit("EfilterClientsMode","active")}}/>
                <input type="button" value="Заблокированные" id={"blocked"} onClick={()=>{voteEvents.emit("EfilterClientsMode","blocked")}}/>
            </div>
        )
    }

}

export default MobileClient;

