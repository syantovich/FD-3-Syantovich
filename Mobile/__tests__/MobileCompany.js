import React from 'react';
import renderer from 'react-test-renderer';
import MobileFilter from "../components/MobileFilter";



test('работа фильтрации', () => {

    const component = renderer.create(
        <MobileFilter />
    );

    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // найдём в вёрстке компонента саму кнопку
    const buttonFilterAll = component.root.findByProps({ id: "all" })
    const buttonFilterActive = component.root.findByProps({ id: "active" })
    const buttonFilterBlocked = component.root.findByProps({ id: "blocked" })

    // и "нажмём" на неё
    buttonFilterAll.props.onClick();
    buttonFilterActive.props.onClick();
    buttonFilterBlocked.props.onClick();
    // получаем уже изменённый снэпшот
    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // "нажмём" кнопку ещё раз
    buttonFilterAll.props.onClick();
    buttonFilterActive.props.onClick();
    buttonFilterBlocked.props.onClick();

    // и получаем окончательный снэпшот
    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

    /*
    // можно эмулировать события, передавая в качестве объекта события то что нам нужно:
    wrapper.find('select').simulate('change', {
        target: { value: "hello" },
    });
    */

});