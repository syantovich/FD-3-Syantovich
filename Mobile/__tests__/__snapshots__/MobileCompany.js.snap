// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа фильтрации 1`] = `
<div>
  <input
    id="all"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    id="active"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    id="blocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
</div>
`;

exports[`работа фильтрации 2`] = `
<div>
  <input
    id="all"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    id="active"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    id="blocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
</div>
`;

exports[`работа фильтрации 3`] = `
<div>
  <input
    id="all"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    id="active"
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    id="blocked"
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
</div>
`;
